### PR TITLE
feat: support prql

### PR DESCRIPTION
**Pull request by ncuwaln**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary
support prql in databend
https://github.com/datafuselabs/databend/issues/14811

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14922)

### CODE REVIEW

**Comment by sundy-li** [comment]
could we support this inside the ast crate ?

**Comment by ncuwaln** [comment]
> could we support this inside the ast crate ?

Ast crate use tokens as parameter now, but prql's token not same as databend sql’s token. so if we want to support this inside ast, maybe we need make string as ast parser's parameter?

**Comment by sundy-li** [comment]
> Ast crate use tokens as parameter now, but prql's token not same as databend sql’s token. so if we want to support this inside ast, maybe we need make string as ast parser's parameter?

I see now, this pr's approach is ok.

File: src/query/sql/src/planner/planner.rs [hunk 1]
```diff
@@ -67,8 +70,31 @@ impl Planner {
...
+            true => {
+                let options = prqlc::Options::default();
+                match prqlc::compile(sql, &options) {
+                    Ok(res) => {
+                        info!("Try convert prql to sql succeed, generated sql: {}", &res);
+                        prql_converted = true;
+                        res
+                    }
+                    Err(e) => {
+                        warn!(
```
Comment (sundy-li) [inline src/query/sql/src/planner/planner.rs:85]: So each time we turn back into the raw SQL, there will be warn msg.

```
set sql_dialect = 'prql';
...

set sql_dialect = 'mysql';  ----  we should not print this msg I think.
```

Comment (ncuwaln) [reply src/query/sql/src/planner/planner.rs:85]: Yes, SetVariable statement still work at prql dialect, just print a warn log.
We cannot check sql statement type before parse, so we cannot check wheater we should print warn log.

**Comment by sundy-li** [comment]
Good to me, let's add some tests.

**Comment by ncuwaln** [comment]
> Good to me, let's add some tests.

done

**Review by sundy-li — APPROVED** [review]

**Comment by snth** [comment]
Thank you!

**Comment by max-sixty** [comment]
Hi from the PRQL team! Excited to see this.

If you'd like to add links to Databend from our website & docs, please feel free to. We did [this](https://github.com/PRQL/prql/pull/3251) for ClickHouse when they started supporting PRQL, so we could pretty much copy that PR..

**Comment by sundy-li** [comment]
> If you'd like to add links to Databend from our website & docs, please feel free to 

@ncuwaln  Could you send a pr to improve the PRQL's website?

BTW @max-sixty  we will write a blog about PRQL with Databend soon.

**Comment by ncuwaln** [comment]
> @ncuwaln Could you send a pr to improve the PRQL's website?
Of course, i will send a pr soon

**Comment by ncuwaln** [comment]
> Of course, i will send a pr soon

I will send a pr after databend doc ready

**Comment by sundy-li** [comment]
document cc @soyeric128

**Comment by soyeric128** [comment]
`Prql` is missing:

Screenshot 2024-03-25 at 6 12 45 PM

**Comment by ncuwaln** [comment]
> `Prql` is missing:

Already send a pr to fix it

https://github.com/datafuselabs/databend/pull/15091

**Comment by sundy-li** [comment]
A blog about databend & prql:  https://www.databend.com/blog/2024-04-03-databend-integrates-prql/

**Pull request merged**